src/nvim/shada.c

```diff
@@ -2216,8 +2222,22 @@ static inline ShaDaWriteResult shada_read_when_writing(ShaDaReadDef *const sd_re
               }
               shada_free_shada_entry(&wms_entry->data);
             }
+          } else {
+            FOR_ALL_BUFFERS(buf) {
+              if (buf->b_ffname != NULL &&
+                  FNAMECMP(entry.data.filemark.fname, buf->b_ffname) == 0) {
+                fmark_T fm;
+                mark_get(buf, curwin, &fm, kMarkBufLocal, (int)entry.data.filemark.name);
+                if (fm.timestamp >= entry.timestamp) {
+                  set_wms = false;
+                  break;
```

Comment:
And maybe here too? I'm not sure.
```suggestion
                  shada_free_shada_entry(&entry);
                  break;
```